code/modules/power/lighting.dm

```diff
@@ -385,6 +385,67 @@
 		icon_state = "runway50"
 		base_state = "runway5"
 
+/obj/machinery/light/traffic_light
+	name = "warning light"
+	desc = "A small light used to warn when shuttle traffic is expected."
+	icon_state = "runway10"
+	base_state = "runway1"
+	fitting = "bulb"
+	brightness = 0.5
+	light_type = /obj/item/light/bulb
+	allowed_type = /obj/item/light/bulb
+	plane = PLANE_NOSHADOW_BELOW
+	on = 0
+	wallmounted = 0
+	removable_bulb = 0
+	var/connected_dock = null
+
+	New()
+		..()
+		if(src.connected_dock)
+			RegisterSignal(GLOBAL_SIGNAL, src.connected_dock, .proc/dock_signal_handler)
```

Comment:
things should not rely on hardcoded signal strings in the map files.

Reply:
what would be a good approach here? for airbridges, adding a subtype seems straightforward. for warning lights, that's a lot of spare types; is that OK?

Reply:
tbqh I'm not sure what the best solution would be, just that this ain't it 🤷

Reply:
implemented these as dedicated subtypes that set `connected_dock`.